contracts/math/Math.sol

```diff
@@ -13,4 +13,12 @@ library Math {
   function min(uint256 _a, uint256 _b) internal pure returns (uint256) {
     return _a < _b ? _a : _b;
   }
+
+  function average(uint256 _a, uint256 _b) internal pure returns (uint256) {
+    // (_a + _b) / 2 can overflow, so we distribute
+    uint256 c = (_a / 2) + (_b / 2) + ((_a % 2 + _b % 2) / 2);
+    assert((_a <= c && c <= _b) || (_a >= c && c >= _b));
```

Comment:
@leonardoalt how do you feel about these sort of assertions? Do you consider them necessary and correct, or do you think they may add too much unnecessary overhead (and gas costs)?